src/domain/usecases/ExportDatasetsUseCase.js

```diff
@@ -0,0 +1,172 @@
+export class ExportDatasetsUseCase {
+	constructor(dhisRepository, xlsxRepository) {
+		this.dhisRepository = dhisRepository;
+		this.xlsxRepository = xlsxRepository;
+	}
+
+	execute($resource, dataSetsIds, headers, locales) {
+		return this.dhisRepository
+			.get($resource, dataSetsIds)
+			.$promise.then(({ dataSets }) => {
+				const pickedTranslations = dataSets.map((dataSet) => ({
+					...dataSet,
+					pickedTranslations: _.intersection(
+						locales,
+						dataSet.translations
+							.filter(
+								(translation) => translation.property === "NAME"
+							)
+							.map(
+								(translation) =>
+									translation.locale.split("_")[0]
+							)
+					),
+				}));
+				const translatedDatasets = pickedTranslations.flatMap(
+					mapDataSetTranslations
+				);
+				const mappedDatasets = this.dhisRepository.getDataSets([
+					...translatedDatasets,
+					...dataSets,
+				]);
+				const dataSetsWithHeaders = mappedDatasets.map((dataSet) => ({
+					...dataSet,
+					headers: headers.find(({ id }) => id === dataSet.id),
+				}));
+
+				return this.xlsxRepository.createFiles(dataSetsWithHeaders);
+			})
+			.then((blobFiles) => {
+				if (blobFiles.length > 1) {
```

Comment:
When selecting data sets that are not exportable (with custom forms), this returns an empty array and the operation does nothing (with no user feedback), it's a bit weird for the user.

Reply:
Now custom data sets are not retrieved